config/v1/types_dns.go

```diff
@@ -53,6 +53,14 @@ type DNSSpec struct {
 	//
 	// +optional
 	PrivateZone *DNSZone `json:"privateZone,omitempty"`
+	// awsPrivateHostedZoneRole contains the ARN of a role that should be assumed when performing
+	// operations on the cluster's private hosted zone specified in the cluster DNS config.
+	//
+	// When left empty, no role should be assumed.
+	//
+	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):iam:[a-z0-9-]+:[0-9]{12}:role\/.*$`
+	// +optional
+	AWSPrivateHostedZoneRole string `json:"awsPrivateHostedZoneRole,omitempty"`
```

Comment:
Should this be made immutable if we don't support Day2? Could this just be in the `.status` field if it's immutable? Is the intent that we may support Day2 operations in the future, hence `.spec` makes sense?

Reply:
Bump